### PR TITLE
DEVPROD-5210 Use separate GRPC url for GRPC cedar connection

### PR DESCRIPTION
**Pull request by hadjri**
DEVPROD-5210

### Description
This adds a new `GRPCBaseURL` field to the cedar config that is to be used when we create a GRPC connection. The `BaseURL` will remain in use for all other HTTP usages. We'll need to add this field to our DB manually once this is approved before we deploy.

### Testing
Tested in staging.

### CODE REVIEW

**Review by zamj — APPROVED** [review]
Looks great, thanks for taking a look!

**Comment by hadjri** [comment]
evergreen retry

**Comment by hadjri** [comment]
evergreen refresh

File: agent/internal/client/base_client.go [hunk 1]
```diff
@@ -317,10 +317,6 @@ func (c *baseCommunicator) GetExpansionsAndVars(ctx context.Context, taskData Ta
 	return &expAndVars, nil
 }
 
-// TaskConflict is a special agent-internal message that the heartbeat uses to
```
[the remark this answers is not in the document]
Comment (hadjri) [reply agent/internal/client/base_client.go:320]: It was unused o I just removed it

**Review by bynn — APPROVED** [review]

**Comment by hadjri** [comment]
Re-requesting review because I added a commit (from this slack [thread](https://mongodb.slack.com/archives/G979SL0HM/p1711655352396539))

**Review by bynn — APPROVED** [review]

**Pull request merged**